src/database/dynamic-update-query.ts

```diff
@@ -0,0 +1,18 @@
+export function buildUpdateQuery(table: string, data: any, where: Map<string, string>, returnUpdatedRow: boolean = false) {
+    const keys = Object.keys(data);
```

Comment:
you can probably move all of this to a builder class `QueryCriteria`. This should be fine too, make sure you handle the corner cases and base cases

Reply:
something like this...

```ts
const criteria = new QueryCriteria()
  .where('id', 1)
  .where('status', 'active', '!=')
  .set('name', 'New Name')
  .set('updated_at', new Date());
  
  console.log(criteria.build());
  
  ```

Reply:
Taken care, written unit test to validate this functionality. utility.test.ts

Reply:
Added builder pattern